packages/charts/src/chart_types/metric/renderer/dom/text.tsx

```diff
@@ -221,36 +241,45 @@ export const MetricText: React.FunctionComponent<{
         )}
       </div>
       <div>
-        <p className="echMetricText__value" style={{ fontSize: `${VALUE_FONT_SIZE[size]}px` }}>
-          {isFiniteNumber(value)
-            ? parts.map(([type, text], i) =>
-                type === 'numeric' ? (
-                  text
-                ) : (
-                  // eslint-disable-next-line react/no-array-index-key
-                  <span key={i} className="echMetricText__part" style={{ fontSize: `${VALUE_PART_FONT_SIZE[size]}px` }}>
-                    {text}
-                  </span>
-                ),
-              )
-            : style.nonFiniteText}
+        <p
+          className="echMetricText__value"
+          style={{
+            fontSize: `${VALUE_FONT_SIZE[size]}px`,
+            textOverflow: isNumericalMetric ? undefined : 'ellipsis',
+          }}
+          title={textParts.map(([, text]) => text).join('')}
+        >
+          {textParts.map(([emphasis, text], i) => {
+            return emphasis === 'small' ? (
+              <span
+                key={`${text}${i}`}
+                className="echMetricText__part"
+                style={{ fontSize: `${VALUE_PART_FONT_SIZE[size]}px` }}
+              >
+                {text}
+              </span>
+            ) : (
+              text
+            );
+          })}
         </p>
       </div>
     </div>
   );
 };
 
-function splitNumericSuffixPrefix(text: string) {
+function splitNumericSuffixPrefix(text: string): ['normal' | 'small', string][] {
```

Comment:
Can you also add the tuple values here?

```suggestion
function splitNumericSuffixPrefix(text: string): [thing1: 'normal' | 'small', thing2: string][] {
```

Reply:
solved in https://github.com/elastic/elastic-charts/pull/1817/commits/05b71181503948465dbf58f03aa82fafa1da14c8